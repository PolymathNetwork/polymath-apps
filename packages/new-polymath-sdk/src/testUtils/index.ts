import Web3PromiEvent from 'web3-core-promievent';
import { GenericContract } from '~/LowLevel/types';
import { PostTransactionResolver } from '~/PostTransactionResolver';

export class MockedContract<T extends GenericContract> {
  public autoResolve: boolean;
  public errorMsg?: string;
  public fakeTxOnePromiEvent = new Web3PromiEvent();
  public fakeTxTwoPromiEvent = new Web3PromiEvent();
  public failureTxPromiEvent = new Web3PromiEvent();

  public fakeTxOne = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxOnePromiEvent.resolve();
      }
      return this.fakeTxOnePromiEvent.eventEmitter;
    };
  });
  public fakeTxTwo = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxTwoPromiEvent.resolve();
      }
      return this.fakeTxTwoPromiEvent.eventEmitter;
    };
  });

  public failureTx = jest.fn(async () => {
    return () => {
      const err = this.errorMsg || 'Test error';
      this.failureTxPromiEvent.reject(new Error(err));
      return this.failureTxPromiEvent.eventEmitter;
    };
  });

  constructor({
    autoResolve = true,
    errorMsg,
  }: { autoResolve?: boolean; errorMsg?: string } = {}) {
    this.autoResolve = autoResolve;
    this.errorMsg = errorMsg;
  }
}

export const getMockTransactionSpec = (
  method: (args: any) => Promise<any>,
  args: any,
  resolver = async () => {}
) => ({
  method,
  args,
  postTransactionResolver: new PostTransactionResolver(resolver),
});
