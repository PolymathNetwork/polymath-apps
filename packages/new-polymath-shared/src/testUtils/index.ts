export * from './factories';

const originalWindow = {
  ...window,
};

interface WindowWithEthereum extends Window {
  ethereum: {
    networkVersion: string;
    _metamask?: {
      isApproved: () => Promise<boolean>;
    };
    enable(): Promise<any>;
  };
}

interface WindowWithWeb3 extends Window {
  web3: {
    version: string;
  };
}

interface MockEthereumBrowserArgs {
  support?: 'legacy' | 'modern';
  options?: {
    networkId: number;
    loaded: boolean;
  };
}

export function mockEthereumBrowser({
  support = 'modern',
  options = {
    networkId: 1,
    loaded: true,
  },
}: MockEthereumBrowserArgs) {
  const { networkId, loaded } = options;

  if (support === 'modern') {
    window = (Object.assign(window, {
      ethereum: {
        networkVersion: loaded ? `${networkId}` : undefined,
      },
    }) as any) as WindowWithEthereum;
  }
  if (support === 'legacy') {
    window = Object.assign(window, {
      web3: {
        version: networkId,
      },
    } as any) as WindowWithWeb3;
  }

  return {
    restore: () => {
      window = originalWindow;
    },
    load: () => {
      if (support === 'modern') {
        (window as WindowWithEthereum).ethereum.networkVersion = `${networkId}`;
      }
      if (support === 'legacy') {
        (window as WindowWithWeb3).web3.version = `${networkId}`;
      }
    },
  };
}
