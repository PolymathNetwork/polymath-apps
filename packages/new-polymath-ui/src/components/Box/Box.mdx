---
name: Box
menu: Primitives
---

import { Playground, PropsTable } from 'docz';

import { Box } from './Box';

# Box

<PropsTable of={Box} />

## Basic usage

<Playground>
  <Box>This is a box.</Box>
</Playground>
