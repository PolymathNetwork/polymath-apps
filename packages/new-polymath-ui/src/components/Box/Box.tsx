import * as React from 'react';
import styled, { StyledProps } from 'styled-components';
import {
  minWidth,
  MinWidthProps,
  maxWidth,
  MaxWidthProps,
  width,
  WidthProps,
  height,
  HeightProps,
  space,
  SpaceProps,
  textAlign,
  TextAlignProps,
} from 'styled-system';

export type BoxProps = StyledProps<any> &
  MinWidthProps &
  MaxWidthProps &
  WidthProps &
  HeightProps &
  SpaceProps &
  TextAlignProps;

export const Box = styled.div<BoxProps>`
  box-sizing: border-box;
  ${minWidth};
  ${maxWidth};
  ${width};
  ${height};
  ${textAlign};
  ${space};
`;

// // TODO @grsmto: remove when https://github.com/pedronauck/docz/issues/337 is resolved
export const BoxDocz = (props: BoxProps) => {
  return <Box {...props} />;
};
