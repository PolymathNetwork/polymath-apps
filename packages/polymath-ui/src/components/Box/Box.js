// @flow

import styled from 'styled-components';
import { width, maxWidth, height, space, textAlign } from 'styled-system';

const Box = styled.div`
  box-sizing: border-box;
  ${width};
  ${maxWidth};
  ${height};
  ${space};
  ${textAlign};
`;

export default Box;
