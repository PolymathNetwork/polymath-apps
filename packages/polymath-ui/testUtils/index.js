import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import theme from '../src/theme';

const middlewares = [thunk];

const customRender = (node, ...opts) => {
  const { rerender, ...result } = render(
    <ThemeProvider theme={theme}>{node}</ThemeProvider>,
    ...opts
  );

  const customRerender = (rerenderNode, ...rerenderOpts) => {
    return rerender(
      <ThemeProvider theme={theme}>{rerenderNode}</ThemeProvider>,
      ...rerenderOpts
    );
  };

  return { rerender: customRerender, ...result };
};

export const mockStore = configureMockStore(middlewares);
export * from 'react-testing-library';
export { customRender as render };
