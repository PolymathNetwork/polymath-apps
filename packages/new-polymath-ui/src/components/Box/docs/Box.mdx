---
name: Box
menu: Primitives
---

import { Playground, PropsTable } from 'docz';

import { Box, BoxDocz } from '../Box';

# Box

<PropsTable of={BoxDocz} />

## Basic usage

<Playground>
  <Box>This is a box.</Box>
</Playground>
