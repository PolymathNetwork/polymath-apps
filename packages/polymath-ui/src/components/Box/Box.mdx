---
name: Grid
---

import { Playground, PropsTable } from 'docz';

import Box from './';

# Grid

<PropsTable of={Box} />

## Basic usage

<Playground>
  <Box>
    This is a box.
  </Box>
</Playground>
