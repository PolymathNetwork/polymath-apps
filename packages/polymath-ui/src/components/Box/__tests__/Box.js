import React from 'react';
import { render } from '../../../../testUtils';

import Box from '../';

test('renders without crashing', () => {
  const { container } = render(<Box>This is a box.</Box>);
  expect(container.firstChild).toMatchSnapshot();
});
