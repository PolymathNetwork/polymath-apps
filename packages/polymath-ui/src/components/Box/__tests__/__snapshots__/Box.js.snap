// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="sc-bdVaJa fxZZEH"
>
  This is a box.
</div>
`;
